### ISSUE TITLE
Option to show repositories in directories

### ISSUE THREAD

**Issue by camilstaps** [klaussilveira/gitlist#786]
This is a feature request.

On [my install](https://git.camilstaps.nl/), I currently have a directory structure like this:

```
<repository root>
  - archived
    - repo1.git
    - repo2.git
  - repo3.git
  - repo4.git
```

The list of repositories on the gitlist homepage looks like:

```
archived/repo1.git
archived/repo2.git
repo3.git
repo4.git
```

I would suggest giving the option (through `config.ini`) to have the repository browser reflect the directory structure. I.e., the homepage could look like this:

```
archived/
repo3.git
repo4.git
```

And clicking on `archived` would bring you to:

```
repo1.git
repo2.git
```

This would make it much easier to structure large sets of repositories.

**Comment by alehaa** [comment]
I think this is related to #391.

**Comment by camilstaps** [comment]
@alehaa it is related in the sense that it is also concerns git repositories further down the directory tree. But if I understand it correctly, it asks that `client/project/solution.git` is shown that way rather than `project/solution.git` on the repository listing. It seems to me it should be tagged bug, not feature. (Comparing to this feature request, I'm suggesting that the repository listing has an (un)foldable directory structure so that in long lists it is easier to get an overview.)

**Comment by alehaa** [comment]
@camilstaps as far as I understand, you'd like to split the repository overview in "folders" and "repositories" - as a new "feature", because at the moment GitList doesn't care about directory structure and just lists all repositories in one list.

However, for deeper hierarchies, the full path of the repository is required - which is addressed by #391. ;)

**Comment by camilstaps** [comment]
Yes, I understand. I created two PRs (as you might have seen):

- #788, very minimal, to resolve #391
- #789, more involved, which resolves the current issue and addresses #391 by doing that
